Doc/library/pathlib.rst

```diff
@@ -946,6 +946,108 @@ call fails (for example because the path doesn't exist).
+   path (which begins with *self*) to a file or directory in *dirpath*, do
+   ``dirpath / name``. Whether or not the lists are sorted depends on the file
+   system.
+
+   If optional argument *top_down* is ``True`` or not specified, the triple for a
+   directory is generated before the triples for any of its subdirectories
+   (directories are generated top-down).  If *top_down* is ``False``, the triple
+   for a directory is generated after the triples for all of its subdirectories
+   (directories are generated bottom-up). No matter the value of *top_down*, the
+   list of subdirectories is retrieved before the tuples for the directory and
+   its subdirectories are generated.
+
+   When *top_down* is True, the caller can modify the *dirnames* list in-place
+   (For example, using :keyword:`del` or slice assignment), and :meth:`Path.walk`
+   will only recurse into the subdirectories whose names remain in *dirnames*;
+   this can be used to prune the search, or to impose a specific order of visiting,
+   or even to inform :meth:`Path.walk` about directories the caller creates or
+   renames before it resumes :meth:`Path.walk` again. Modifying *dirnames* when
+   *top_down* is False has no effect on the behavior of :meth:`Path.walk()`, since the
+   directories in *dirnames* have already been generated by the time *dirnames*
+   is yielded to the caller.
+
+   By default errors from :func:`os.scandir` call are ignored.  If
+   optional argument *on_error* is specified, it should be a callable; it
+   will be called with one argument, an :exc:`OSError` instance. It can
+   report the error to continue with the walk, or raise the exception
+   to abort the walk. Note that the filename is available as the
+   ``filename`` attribute of the exception object.
+
+   By default, :meth:`Path.walk` will not walk down into symbolic links that
+   resolve to directories.  Set *follow_symlinks* to ``True`` to visit directories
+   pointed to by symlinks, on systems that support them.
+
+   .. note::
+
+      Be aware that setting *follow_symlinks* to ``True`` can lead to infinite
+      recursion if a link points to a parent directory of itself. :meth:`Path.walk`
+      does not keep track of the directories it visited already.
+
+   .. note::
+      This method assumes that the current working directory is not changing while
+      walking relative paths.
+   .. note::
+
+      :meth:`Path.walk` assumes the directories have not been modified between
+      its resumptions. For example, if a directory from *dirnames* has been replaced
+      with a symlink and *follow_symlinks* = ``False``, :meth:`Path.walk` will
+      still try to descend into it. To prevent such behavior, remove directories
+      from *dirnames* if they have been modified and you do not want to
+      descend into them anymore.
+
+   .. note::
+
+      Unlike :func:`os.walk`, :meth:`Path.walk` adds symlinks to directories into *filenames*
+      if *follow_symlinks* is ``True``
```

Comment:
```suggestion
      Unlike :func:`os.walk`, :meth:`Path.walk` lists symlinks to directories into
      *filenames* if *follow_symlinks* is ``True``.
```

Is this a good idea? Mixing directory names and file names simply because they are a symlink seems a bit dangerous. Is there a reason why a directory symlink ends up in *filenames* instead of *dirnames*?

Reply:
Looks like a typo in my docs. Will check it right now.

Reply:
It is a typo. It was supposed to be "is False"